### PR TITLE
Configurable RocksDB info logs

### LINKED ISSUE #2216: Restate RocksDB log files sometimes hang around for a long time pushing disk usage up
We've observed an idle environment in Restate Cloud consuming 100s of MBs even post restart. The files look like this:

```
> kubectl exec restate-0 -- find /restate-data -name '*LOG.old*' -exec ls -lh {} \;
-rw-r--r--. 1 1000 2000 197M Oct  3 10:49 /restate-data/restate-0/local-metadata-store/LOG.old.1727952596344409
-rw-r--r--. 1 1000 2000 209M Oct  3 10:49 /restate-data/restate-0/db/LOG.old.1727952596237953
-rw-r--r--. 1 1000 2000 203M Oct  3 10:49 /restate-data/restate-0/local-loglet/LOG.old.1727952596692048
```

Restarting the process clears the original files but new ones remain:

```
> kubectl exec restate-0 -- find /restate-data -name '*LOG.old*' -exec ls -lh {} \;
-rw-r--r--. 1 1000 2000 78M Nov  3 07:48 /restate-data/restate-0/local-metadata-store/LOG.old.1730620202653580
-rw-r--r--. 1 1000 2000 84M Nov  3 07:48 /restate-data/restate-0/db/LOG.old.1730620202546276
-rw-r--r--. 1 1000 2000 81M Nov  3 07:48 /restate-data/restate-0/local-loglet/LOG.old.1730620202879822

> kubectl exec restate-0 -- du -csh /restate-data/
242M	/restate-data/
242M	total
```

Same environment, a couple of days later:

```
> kubectl exec restate-0 -- find /restate-data -name '*LOG.old*' -exec ls -lh {} \;
-rw-r--r--. 1 1000 2000 78M Nov  3 07:48 /restate-data/restate-0/local-metadata-store/LOG.old.1730620202653580
-rw-r--r--. 1 1000 2000 84M Nov  3 07:48 /restate-data/restate-0/db/LOG.old.1730620202546276
-rw-r--r--. 1 1000 2000 81M Nov  3 07:48 /restate-data/restate-0/local-loglet/LOG.old.1730620202879822

> kubectl exec restate-0 -- du -csh /restate-data/
260M	/restate-data/
260M	total
```

Note the files remain unchanged.

### PR DESCRIPTION
**Pull request by muhamadazmy**
Configurable RocksDB info logs

Fixes #2216

### CODE REVIEW

**Review by pcholakov — APPROVED** [review]
Thanks @muhamadazmy, looks good! 👍

File: crates/types/src/config/rocksdb.rs [hunk 1]
```diff
@@ -76,6 +76,38 @@ pub struct RocksDbOptions {
     /// Default: "except-detailed-timers"
     #[serde(skip_serializing_if = "Option::is_none")]
     rocksdb_statistics_level: Option<StatisticsLevel>,
+
+    /// # RocksDB log level
+    ///
+    /// Verbosity of the LOG.
+    rocksdb_log_level: Option<RocksDbLogLevel>,
```
Comment (AhmedSoliman) [inline crates/types/src/config/rocksdb.rs:83]: Can you mention the default value in docs?

File: crates/types/src/config/rocksdb.rs [hunk 2]
```diff
@@ -76,6 +76,38 @@ pub struct RocksDbOptions {
...
+
+    /// # RocksDB log keep file num
+    ///
+    /// Number of info LOG files to keep
+    rocksdb_log_keep_file_num: Option<usize>,
+
+    /// # RocksDB log max file size
+    ///
+    /// Max size of info LOG file
```
Comment (AhmedSoliman) [inline crates/types/src/config/rocksdb.rs:92]: ditto ("64MB" is the default)

File: crates/types/src/config/rocksdb.rs [hunk 3]
```diff
@@ -76,6 +76,38 @@ pub struct RocksDbOptions {
...
+    #[cfg_attr(feature = "schemars", schemars(with = "Option<ByteCount>"))]
+    rocksdb_log_max_file_size: Option<NonZeroByteCount>,
```
Comment (AhmedSoliman) [inline crates/types/src/config/rocksdb.rs:94]: ditto

File: crates/types/src/config/rocksdb.rs [hunk 4]
```diff
@@ -76,6 +76,38 @@ pub struct RocksDbOptions {
...
+
+    /// # RocksDB log level
+    ///
+    /// Verbosity of the LOG.
+    rocksdb_log_level: Option<RocksDbLogLevel>,
+
+    /// # RocksDB log keep file num
+    ///
+    /// Number of info LOG files to keep
+    rocksdb_log_keep_file_num: Option<usize>,
```
Comment (AhmedSoliman) [inline crates/types/src/config/rocksdb.rs:88]: ditto

**Review by AhmedSoliman — APPROVED** [review]
Thanks for fixing this. Just a few nits and we should be good to merge.

**Pull request merged**